catalog/ui/catalog-ui-search/src/main/webapp/js/widgets/cesium.polygon.js

```diff
@@ -34,36 +35,37 @@ define([
                 this.drawPolygon(this.model);
             },
             drawPolygon: function(model) {
-                var polygonPoints = model.toJSON().polygon;
-                if (!polygonPoints) {
-                    return;
-                }
-                var setArr = polygonPoints;
-                if (setArr.length < 3) {
-                    return;
-                }
-                if (polygonPoints[0].toString() !== polygonPoints[polygonPoints.length - 1].toString()) {
-                    polygonPoints.push(polygonPoints[0]);
-                }
+                var json = model.toJSON();
+                var isMultiPolygon =  ShapeUtils.isArray3D(json.polygon);
+                var polygons = isMultiPolygon ? json.polygon : [ json.polygon ];
+
+                var color = this.model.get('color');
 
                 // first destroy old one
                 if (this.primitive && !this.primitive.isDestroyed()) {
                     this.options.map.scene.primitives.remove(this.primitive);
                 }
 
-                var color = this.model.get('color');
-
                 this.primitive = new Cesium.PolylineCollection();
-                this.primitive.add({
-                    width: 8,
-                    material: Cesium.Material.fromType('PolylineOutline', {
-                        color: color ? Cesium.Color.fromCssColorString(color) : Cesium.Color.KHAKI,
-                        outlineColor: Cesium.Color.WHITE,
-                        outlineWidth: 4
-                    }),
-                    id: 'userDrawing',
-                    positions: Cesium.Cartesian3.fromDegreesArray(_.flatten(setArr))
-                });
+
+                (polygons || []).forEach(function(polygonPoints){
```

Comment:
:pencil2: It seems safer to ensure that `polygons` always contains a list rather than a list or `undefined` and checking it here; maybe replace `(polygons || [])` with `polygons` and change `json.polygon` to `(json.polygon || [])` in the above declaration of `polygons`?